apps/dapp/src/components/Pages/Core/DappPages/Dashboard/Table/TxnHistoryTable.tsx

```diff
@@ -31,117 +38,166 @@ enum DebtToken {
-      ]
+      width: '10.87%',
+      rowFilter: {
+        filterFn: useDebouncedCallback(async (event: HTMLInputElement) => {
+          setRowFilter((s) => ({ ...s, token: event.value }));
+        }, 200),
+        // TODO: get dropdown values programatically, see https://github.com/TempleDAO/temple/pull/880#discussion_r1386151604
+        dropdownOptions: [
+          { label: 'All', value: undefined },
+          { label: DebtToken.DAI, value: DebtToken.DAI },
+          { label: DebtToken.TEMPLE, value: DebtToken.TEMPLE },
+        ],
+        defaultValue: { label: 'All', value: undefined },
+      },
     },
     {
       name: TableHeaders.Amount,
       orderDesc: undefined,
+      width: '12.93%',
     },
     {
       name: TableHeaders.TxHash,
       orderDesc: undefined,
+      width: '17.13%',
     },
   ]);
 
-  const updateTableHeadersOrder = (clickedHeader: TxHistoryTableHeader) => setTableHeaders(prevState => {
-    const newState = prevState.map((prevStateHeader)=>{
-      if(prevStateHeader.name === clickedHeader.name){
-        return { ...prevStateHeader, orderDesc: !prevStateHeader.orderDesc };
-      }
-      else {
-        return { ...prevStateHeader, orderDesc: undefined};
-      }
+  const updateTableHeadersOrder = (clickedHeader: TxHistoryTableHeader) =>
+    setTableHeaders((prevState) => {
+      const newState = prevState.map((prevStateHeader) => {
+        if (prevStateHeader.name === clickedHeader.name) {
+          return { ...prevStateHeader, orderDesc: !prevStateHeader.orderDesc };
+        } else {
+          return { ...prevStateHeader, orderDesc: undefined };
+        }
+      });
+      return newState;
     });
-    return newState;
-  });
-  
+
+  useEffect(() => {
+    const selectedStrategy = dashboardTypeToStrategyKey(dashboardType);
+    setTableHeaders((prevState) => {
+      const newState = prevState.map((prevStateHeader) => {
+        if (prevStateHeader.name === TableHeaders.Strategy) {
+          // When user changes dashboard url:
+          //  1. reset page
+          setCurrentPage(1);
+          //  2. update table strategy dropdown default value
+          return {
+            ...prevStateHeader,
+            rowFilter: prevStateHeader.rowFilter && {
+              filterFn: prevStateHeader.rowFilter.filterFn,
+              dropdownOptions: prevStateHeader.rowFilter.dropdownOptions,
+              defaultValue: { label: selectedStrategy, value: selectedStrategy },
+            },
+          };
+        }
+        return prevStateHeader;
+      });
+      return newState;
+    });
+  }, [dashboardType]);
+
   const availableRows = useTxHistoryAvailableRows({
     dashboardType,
     txFilter,
-    rowFilter
+    rowFilter,
   });
 
+  // Only change the blockNumber when the page is refreshed
+  // it ensures consistency in subsequent pagination queries results
+  if (blockNumber === 0 && availableRows.data) setBlockNumber(availableRows.data.blockNumber);
```

Comment:
I don't think it's a big deal for the charts or metrics because once data is loaded, there are not many user interactions. In pagination is different because the first page can show data from block 1, and then next page from block 2, which could lead to data incongruity, but maybe I'm wrong?